src/lib/libpthread.js

```diff
@@ -427,14 +408,50 @@ var LibraryPThread = {
             createScriptURL: (ignored) => new URL("{{{ TARGET_JS_NAME }}}", import.meta.url)
           }
         );
-        worker = new Worker(p.createScriptURL('ignored'), workerOptions);
+        worker = new Worker(p.createScriptURL('ignored'), {
+#if EXPORT_ES6
+            'type': 'module',
+#endif
+#if ENVIRONMENT_MAY_BE_NODE
+            // This is the way that we signal to the node worker that it is hosting
+            // a pthread.
+            'workerData': 'em-pthread',
+#endif
+#if ENVIRONMENT_MAY_BE_WEB || ENVIRONMENT_MAY_BE_WORKER
+            // This is the way that we signal to the Web Worker that it is hosting
+            // a pthread.
+#if ASSERTIONS
+            'name': 'em-pthread-' + PThread.nextWorkerID,
```

Comment:
Previously @sbc100 you had kindly suggested making use of a macro.
```
{{{
globalThis.WORKER_OPTIONS = {
   ...
};
null;
}}}
```

This would have been a great solution, however, because we use a dynamic `PThread.nextWorkerID` reference I couldn't think of a way to make it work. Let me know if you have any other ideas.

Reply:
I'm pretty sure we can still find a way to avoid these 4-fold duplication.    I'll see if I can some up with something

Reply:
Is it possible to capture a raw string and then output that in these templates? Is there a particular templating framework being used? 🤔